src/integrations/misc/extract-text.ts

```diff
@@ -22,6 +23,10 @@ export async function extractTextFromFile(filePath: string): Promise<string> {
 		default:
 			const isBinary = await isBinaryFile(filePath).catch(() => false)
 			if (!isBinary) {
+				const fileSize = await getFileSizeInKB(filePath)
+				if (fileSize > 300) {
+					throw new Error(`File is too large to read into context.`)
+				}
```

Comment:
I think https://github.com/RooVetGit/Roo-Code/pull/1926 will end up being a better solution for this (it will respect the maxReadFileLine setting)